frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx

```diff
@@ -550,6 +542,17 @@ class Visualization extends PureComponent {
   }
 }
 
+function checkIsLoading(series, props) {
```

Comment:
This code has already been far from being ideal, but I think this change complicates it even further. Previously, missing data the sign the card was loading and now we are introducing another flag for that without replacing the old behavior.

The original issue can be fixed by removing stale data for all cards even on other tabs when filters change. We do not need stale data at all, the original issue is about wrongly showing it. At the same time, it would not add an additional flag `isLoading` and instead rely on the existing behavior.

@oleggromov how does this sound to you?

Reply:
This sounds good although I suspect it may break even more tests. I'll try now